builder/parser/parser.go

```diff
@@ -85,10 +85,7 @@ func parseLine(line string) (string, *Node, error) {
 		return "", nil, err
 	}
 
-	if sexp.Value != "" || sexp.Next != nil || sexp.Children != nil {
-		node.Next = sexp
-	}
-
+	node.Next = sexp
```

Comment:
This `if` was here to check for the case of `[]` turning into `[""]` and turning it back into `[]`.  By fixing `parseJSON` to return `nil` (the linked-list representation of an empty list), this `if` is no longer required, so now `[]` and `[""]` can both round-trip successfully.